yarn-project/simulator/src/avm/avm_simulator.test.ts

```diff
@@ -17,26 +18,48 @@ import {
   initExecutionEnvironment,
   initGlobalVariables,
   initL1ToL2MessageOracleInput,
+  initMachineState,
 } from './fixtures/index.js';
-import { Add, CalldataCopy, Return } from './opcodes/index.js';
+import { Add, CalldataCopy, Instruction, Return } from './opcodes/index.js';
 import { encodeToBytecode } from './serialization/bytecode_serialization.js';
 
 describe('AVM simulator: injected bytecode', () => {
-  it('Should execute bytecode that performs basic addition', async () => {
-    const calldata: Fr[] = [new Fr(1), new Fr(2)];
+  let calldata: Fr[];
+  let ops: Instruction[];
+  let bytecode: Buffer;
 
-    // Construct bytecode
-    const bytecode = encodeToBytecode([
+  beforeAll(() => {
```

Comment:
Do you plan to add more tests here? How detailed do we want to be in checking that gas costs are set properly?

Reply:
I'm planning on adding more tests for covering the more "dynamic" gas cost functions, for checking reverts, and for testing the different type of gases.